product_category_code_unique/models/res_config_settings.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2023 ACSONE SA/NV
+# Copyright 2024 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+
+    _inherit = "res.config.settings"
+
+    product_cat_code_unique_restriction = fields.Selection(
```

Comment:
The v15 module only had 1 option: "system". I think that option should be the default one

Reply:
But the module shoud behave the same way as before without changing any setting, with this improvement the default behaviour will change

Reply:
The default behaviour has changed because the module has changed. I don't think we should set a default restriction in a module that accepts different types of restrictions, as we don't know which restrictions the user wants to use. A default restriction could throw exceptions that the user does not want.